libs/lensfun/lens.cpp

```diff
@@ -1115,7 +1115,7 @@ static float __vignetting_dist (
     float d1 = 0.1 / distance;
     float d2 = 0.1 / x.Distance;
 
-    return sqrt (square (f2 - f1) + square (a2 - a1) + square (d2 - d1));
+    return sqrt (pow(f2 - f1, 2) + pow (a2 - a1, 2) + pow (d2 - d1, 2));
```

Comment:
Potentially that may affect performance. `std::pow` is a common algorithm with lots of additional checks, unless it is not optimized by the compiler. And compilers do strange things - gcc and clang are good, but MSVC does not optimize it at all and always call `pow` or `powf` function. Moreover MSVS and clang distinguish `std::pow(x, 2)` and `std::pow(x, 2.f)` and in the first case it casts everything to `double` which is even slower.

There is an `std::hypot` function, which is more precise:
```suggestion
    return hypot(f2 - f1, a2 - a1, d2 - d1);
```
So if we do care about precision, than is should be replaced with `hypot`, but if we care about performance, the old `x*x` can be better than `std::pow(x, 2)`.

Reply:
Doesn't look performance critical. Running the tests with coverage the function got 481 hits.